homeassistant/helpers/trigger.py

```diff
@@ -68,6 +73,105 @@ class TriggerInfo(TypedDict):
     trigger_data: TriggerData
 
 
+@dataclass
+class PluggableActionsEntry:
+    """Holder to keep track of all plugs and actions for a given trigger."""
+
+    plugs: set[PluggableAction] = field(default_factory=set)
+    actions: dict[object, tuple[HassJob, dict[str, Any]]] = field(default_factory=dict)
+
+
+class PluggableAction:
+    """A pluggable action handler."""
+
+    _trigger: dict[str, str]
+    _entry: PluggableActionsEntry | None = None
+
+    def __init__(self, update: CALLBACK_TYPE | None = None) -> None:
+        """Initialize."""
+        self._update = update
```

Comment:
What's the purpose of the update callback? Let's write that in the docstring.

Reply:
Support flags in the entity is adjusted based on availability of turn on hook. If no hook is available, turn on ability is removed.